src/Gren/Kernel/HttpServer.js

```diff
@@ -0,0 +1,97 @@
+/*
+
+import Gren.Kernel.Scheduler exposing (binding, succeed, fail, rawSpawn)
+import HttpServer exposing (CreateServerError, toRequest)
+import Platform exposing (sendToApp, sendToSelf)
+
+*/
+
+const http = require("http");
+
+var _HttpServer_subs = new WeakMap();
```

Comment:
Here I would do things a little differently. In general, I want kernel code to be as free of state as possible.

I believe that passing a request listener to `http.createServer` is optional. Instead, you could use `.on` and `.off` to set/unset a single listener depending on the number of subscribers. That listener can be a closure containing the router.

I would also use `sendToSelf` in order to relay the message to the effect module, and then have the effect module communicate with the app.

Reply:
Agree with all of this. Not sure how to get the `router` and `msg` needed to send to self if I don't track the subs on the js side though. Let me know if you have thoughts on that. Either way, I should have some time to give it a shot tonight.

Reply:
Oh right. Listener gets attached when the sub is created. The js part of my brain is weak 😅

I can picture how this should work now. Thanks 👍

Reply:
Nothing about effect modules and their implementation is straight forward, happy to help :)